### ISSUE TITLE
Preview: sync preview tempo to primary deck tempo, button to toggle?

### ISSUE THREAD

**Issue by mxmilkiib** [mixxxdj/mixxx#12615]
### Feature Description

A control and button near the Preview deck that syncs the tempo of its playback with that of the main playing tempo would be really handy.

**Comment by ronso0** [comment]
What does that help with?
Wouldn't we also need a keylock toggle? Or lock it to enabled?

FYI you can map `[PreviewDeckN],beatsync` to the keyboard.

**Comment by LamprechtHase** [comment]
I would find that very useful indeed.
I tried the mapping of beat sync to the keyboard. It works, but is inconvenient, and barely useful without key lock.
Buttons could be stuffed into the preview deck as illustrated. This would truncate the track info though.

<img width="508" height="116" alt="Image" src="https://github.com/user-attachments/assets/41c659ad-7191-420f-b933-33ed444377c0" />

I find the preview deck as is barely useful. It yields a terrible cacophony with what is playing on the main decks.

**Comment by ronso0** [comment]
Hmm, with the minimized Sync button from the samplers it could work <img width="34" height="20" alt="Image" src="https://github.com/user-attachments/assets/00e9f731-4b1a-46b4-9463-391b1febadb3" />

Though there is no BPM display or tempo fader and no key=original display, so we'd need to merge those into the Sync and Keylock buttons.
What about:
* Sync
   * off, tempo 100%: default, grey
   * off, tempo offset: default + colored border
   * on: blue
   * right-click: reset tempo
* Keylock
   * basically the same scheme for off | off + key off | on
   * right-click: reset key
